pytest_rerunfailures.py

```diff
@@ -106,6 +115,86 @@ class RerunPlugin(object):
 
     def __init__(self):
         self.tests_to_rerun = set([])
+        self.testrun_stats = {}
+        self.rerun_stats = {
```

Comment:
Not really clear why do we need two different variables - `testrun_stats` and `rerun_stats`? Also names are very similar which makes it confusing.

Reply:
`self.testrun_stats` - it is stats for `main (planned)` execution so we able to track amount of tests failed and theirs traces.
`self.rerun_stats` - it is stats for `additional (rerun)` execution and will be saved as report. Since `self.rerun_stats` and `self.testrun_stats` share the same `test.nodeid` we could use `self.testrun_stats` for tracking `flake`\`failure` status for all possible reruns and then populate original traces to them.